app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -4207,16 +4219,55 @@ class BrowserTabViewModel @Inject constructor(
         }
     }
 
-    override fun addDocumentStartJavaScript(webView: WebView) {
+    fun configureWebView(
+        webView: DuckDuckGoWebView,
+        callback: WebViewCompatMessageCallback?,
+    ) {
         viewModelScope.launch {
-            addDocumentStartJavascriptPlugins.getPlugins().forEach {
-                it.addDocumentStartJavaScript(
-                    webView,
-                )
+            addDocumentStartJavaScript(webView)
+
+            callback?.let {
+                webMessagingPlugins.getPlugins().forEach { plugin ->
+                    plugin.register(callback, webView)
+                }
             }
         }
     }
 
+    fun postContentScopeMessage(
+        eventData: SubscriptionEventData,
+        webView: WebView,
+    ) {
+        viewModelScope.launch {
+            postMessageWrapperPlugins
+                .getPlugins()
+                .firstOrNull { it.context == "contentScopeScripts" }
+                ?.postMessage(eventData, webView)
+        }
+    }
+
+    private suspend fun addDocumentStartJavaScript(webView: WebView) {
+        addDocumentStartJavascriptPlugins.getPlugins().forEach {
+            it.addDocumentStartJavaScript(
+                webView,
+            )
+        }
+    }
+
+    suspend fun privacyProtectionsUpdated(webView: WebView) {
+        if (withContext(dispatchers.io()) { !androidBrowserConfig.updateScriptOnPageFinished().isEnabled() }) {
+            addDocumentStartJavascriptPlugins
+                .getPlugins()
+                .filter { plugin ->
+                    (plugin.context == "contentScopeScripts")
+                }.forEach {
+                    it.addDocumentStartJavaScript(webView)
+                }
+        } else {
+            addDocumentStartJavaScript(webView)
+        }
+    }
+
```

Comment:
Struggling to quickly parse this function. My understanding is that we call `addDocumentStartJavascriptPlugins.getPlugins()` in either case. but in one case we filter to only have `context = contentScopeScripts`. And that's all gated on whether `androidBrowserConfig.updateScriptOnPageFinished` is enabled or not. 

I've flagged just for you to double it's doing the correct thing since I'm not able to fully verify it is, but i'm happy if you say that's all working as intended. and if you spot any way to make it clearer what's happening, all the better.